<div class="alert alert-info"> Hey! This is a continual work in progress, which means things are bound to be broken! </div>

# Exploring the PLAICRAFT Datasets
Liam Murray (87357539) | May 21, 2025

<span style="color: grey">Introductory activities and workflows for approaching the PLAICRAFT dataset.</span>

## Activity #1 - Preliminary

**Q: What is the dataset about?** \
`players.csv` provides demographic information about players, including Name, Age, Gender, Experience etc. \
`sessions.csv` provides information on each recorded game session, including the player (via hashed email), and the start and end times (both the unix time stamp and the adjusted calendar date and time). 

**Q: How many variables are there?** \
A quick call of the `head` function shows that there are 5 variables contained within `sessions.csv` and 7 variables within `players.csv` when presented as a tibble.

In [25]:
library(tidyverse)
require(janitor)
library(lubridate)
library(hms)

session_data <- read_csv("data/sessions.csv") |> 
    clean_names() |>
    mutate(session_length = (original_end_time - original_start_time)/60)
player_data <- read_csv("data/players.csv") |> 
    clean_names() |>
    mutate(gender = as_factor(gender)) |>
    mutate(experience = as_factor(experience))

head(session_data)
head(player_data)

session_data

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


hashed_email                                                    
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12      
  session_length
1 0             
2 0             
3 0             
4 0             
5 0             
6 0

experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashed_email                                                     played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

hashed_email                                                    
1    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2    36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3    f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5    36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
7    fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
8    ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
9    96e190b0bf3923cd8d349eee467c09d1130af143335779251492eb4c2c058a5f
10   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
11   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
12   f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
13   f2826fb8dbce4d450348f99cb27ade184b713998d9679780442efaaf218038f2
14   b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
15   24d4892c2c3ce11d3e54c3bf31ee218901cac7a2f564911d67424c8fdf0874fe
16   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
17   88247d9a46fc214a12485dcbcbb03a8ddebfe8c1ec5fe209cd660147625c8e62
18   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
19   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
20   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
21   bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
22   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
23   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
24   ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
25   ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
26   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
27   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
28   bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
29   b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
30   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
⋮    ⋮                                                               
1506 ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
1507 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1508 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1509 b3510c708bd50bf9f75e6e02bb6fe14edb705e0ea671ee769a45c613750e8b58
1510 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1511 e74c60a92c0100e7240be56d66969db85856152b048c639331a3595f901461ae
1512 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1513 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1514 e74c60a92c0100e7240be56d66969db85856152b048c639331a3595f901461ae
1515 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1516 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1517 a1e0438d568b9c05a93cadaf99969e4d44794d19dea31edd723cbc916f878c33
1518 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1519 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
1520 55d24216db39c27e1f17cc43d3127cbf8ed76ada6d098202b53ded319855e2c1
1521 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1522 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1523 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1524 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1525 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1526 18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3
1527 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1528 18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3
1529 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1530 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1531 36d9cbb4c6

**Q: How many observations are there?** \
There are 196 observations within `players.csv` and 1535 observations within `sessions.csv`.

In [26]:
summarize(
    player_data, num_players = n()
)

summarize(
    session_data, num_sessions = n()
)

num_players
1 196

num_sessions
1 1535

## Activity #2 - Outcome Variable

<div class="alert alert-warning"> Tip: Think about how you are organising your workbook: add more code and markdown cells (and arranged them!) to keep your notebook neat</div>

**Q:Identify the main outcome/categorical/label variable in the dataset.**

Outcome: Subscribe - this could be reflective of question 1 (characteristics/behaviours being predictive of subscribing to a newsletter).
         Played_hours - this could be an outcome when analyzing which players are most likely to contribute time to the server.

Categorical: Age, Gender, and Experience - these could be used as the predictors for behaviours within players, including likelyhood to subscribe, activity on the server, and gaming patterns throughout the day.

Label: Name, hashedEmail - these variables are used to identify and track individual participants, these would likely not be useful for modeling and were only collected for participant registration.

**Q:How many values/groups are in this variable?**
<center>

| **Variable** | **# of groups** |
|--------------|-----------------|
| experience   | 5               |
| subscribe    | 2               |
| played_hours | 43              |
| gender       | 7               |
| age          | 31              |

</center>

**Q:How many observations are there in each value/group?**

Played hours had a large variation within the number of hours played and how many players, and was excluded from this section due to the complexity of presenting this information in an accessible and meaningful way.

<img src="Visuals/Experience-Levels.png" alt="Player Experience Levels" width="300"/>

<img src="Visuals/Subscription-Status.png" alt="Player Subscription Status" width="300"/>

<img src="Visuals/Gender-Identity.png" alt="Player Subscription Status" width="300"/>
<img src="Visuals/Player-Ages.png" alt="Player Subscription Status" width="300"/>








In [27]:
# player_data |>
#     select(experience,subscribe,played_hours,gender,age) |>
#     group_by(age) |>
#     summarize(count = n()) |>
#     ggplot(aes(x=reorder(age, -count), y=count)) +
#         geom_histogram(stat="identity") +
#         labs(x="Age", y="Number of players") +
#         ggtitle("Age of players")


## Activity #3 - Visualizations

<div class="alert alert-warning">Tip: Try using a range of box plots, scatterplots, bar charts, line graphs, etc.</div>

**Vizualize: What does the distribution of the variable look like?**

**Vizualize: What relationship does it have with some of the other variables?**

ERROR: Error: object 'merged_data' not found
